CppCoreGuidelines.md

```diff
@@ -9550,6 +9550,28 @@ Instead, use a local variable:
 
 * (Moderate) Warn if an object is allocated and then deallocated on all paths within a function. Suggest it should be a local `auto` stack object instead.
 * (Simple) Warn if a local `Unique_pointer` or `Shared_pointer` is not moved, copied, reassigned or `reset` before its lifetime ends.
+Exception: Do not produce such a warning on a local `Unique_pointer` to an unbounded array. (See below.)
```

Comment:
minor, but maybe use unique_ptr instead of Unique_pointer?  someone may think Unique_ptr is an stl type